File: OneCodeTeam/Replicator-activity-17850/[C#]-Replicator activity (CSSharepointrReplicatorActivity)/C#/CSSharePointReplicatorActivity/AssociationData.cs
/****************************** Module Header ******************************\
* Module Name:    AssociationData.cs
* Project:        CSSharePointReplicatorActivity
* Copyright (c) Microsoft Corporation
*
* This class is used to store Association Data.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\*****************************************************************************/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSSharePointReplicatorActivity.MyWorkflow
{
    [Serializable]
    public class AssociationData
    {
        private Contacts partners = default(Contacts);
        public Contacts Partners
        {
            get
            {
                return this.partners;
            }
            set
            {
                this.partners = value;
            }
        }

        private string description = default(string);
        public string Description
        {
            get
            {
                return this.description;
            }
            set
            {
                this.description = value;
            }
        }

        public void AddContact(string contact)
        {
            if (this.partners == null)
            {
                this.partners = new Contacts();
            }
            this.partners.AddContact(contact);
        }
        public string[] Getpartners()
        {
            return this.partners.ContactList.ToArray();
        }
    }

    [Serializable()]
    public class Contacts
    {
        private List<string> contacts;

        public List<string> ContactList
        {
            get { return contacts; }
            set { contacts = value; }
        }
        public void AddContact(string contact)
        {
            if (this.contacts == null)
            {
                this.contacts = new List<string>();
            }
            this.contacts.Add(contact);
        }

    }
}

File: OneCodeTeam/Replicator-activity-17849/[VB]-Replicator activity (VBSharepointrReplicatorActivity)/VB/VBSharePointrReplicatorActivity/AssociationData.vb
'****************************** Module Header ******************************\
' Module Name:    AssociationData.vb
' Project:        VBSharePointReplicatorActivity
' Copyright (c) Microsoft Corporation
'
' This class is used to store Association Data.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

<Serializable()> _
Public Class AssociationData
    Private m_partners As Contacts = Nothing
    Public Property Partners() As Contacts
        Get
            Return Me.m_partners
        End Get
        Set(ByVal value As Contacts)
            Me.m_partners = value
        End Set
    End Property

    Private m_description As String = Nothing
    Public Property Description() As String
        Get
            Return Me.m_description
        End Get
        Set(ByVal value As String)
            Me.m_description = value
        End Set
    End Property

    Public Sub AddContact(ByVal contact As String)
        If Me.m_partners Is Nothing Then
            Me.m_partners = New Contacts()
        End If
        Me.m_partners.AddContact(contact)
    End Sub
    Public Function Getpartners() As String()
        Return Me.m_partners.ContactList.ToArray()
    End Function
End Class

<Serializable()> _
Public Class Contacts
    Private contacts As List(Of String)

    Public Property ContactList() As List(Of String)
        Get
            Return contacts
        End Get
        Set(ByVal value As List(Of String))
            contacts = value
        End Set
    End Property
    Public Sub AddContact(ByVal contact As String)
        If Me.contacts Is Nothing Then
            Me.contacts = New List(Of String)()
        End If
        Me.contacts.Add(contact)
    End Sub

End Class